pkg/trace/api/api.go

```diff
@@ -221,30 +240,37 @@ func (r *HTTPReceiver) handleTraces(v Version, w http.ResponseWriter, req *http.
 		return
 	}
 
-	// We successfully decoded the payload
 	r.replyTraces(v, w)
 
-	// We parse the tags from the header
-	tags := info.Tags{
+	ts := r.Stats.GetTagStats(info.Tags{
 		Lang:          req.Header.Get("Datadog-Meta-Lang"),
 		LangVersion:   req.Header.Get("Datadog-Meta-Lang-Version"),
 		Interpreter:   req.Header.Get("Datadog-Meta-Lang-Interpreter"),
 		TracerVersion: req.Header.Get("Datadog-Meta-Tracer-Version"),
-	}
+	})
 
-	// We get the address of the struct holding the stats associated to the tags
-	ts := r.Stats.GetTagStats(tags)
+	atomic.AddInt64(&ts.TracesReceived, int64(len(traces)))
+	atomic.AddInt64(&ts.TracesBytes, int64(req.Body.(*LimitedReader).Count))
+	atomic.AddInt64(&ts.PayloadAccepted, 1)
 
-	bytesRead := req.Body.(*LimitedReader).Count
-	if bytesRead > 0 {
-		atomic.AddInt64(&ts.TracesBytes, int64(bytesRead))
-	}
+	r.wg.Add(1)
```

Comment:
I bet that this `wg` of a single goroutine is to allow `handleTraces` to return while `processTraces` isn't done yet?
Worth a code comment!

Reply:
This is here to make sure that we are done processing whatever is "in flight" when shutting down. This is the most common use case for waitgroups as far as I know: graceful shutdown of goroutines. Do you still think it's worth a comment?

Reply:
I guess it is still worth yes, as it wasn't clean to me when I read it.